src/main/java/com/example/chat/groupchatbackend/controllers/ConversationController.java

```diff
@@ -39,5 +36,41 @@ public ResponseEntity createChannel(@RequestBody String name) {
                 .status(HttpStatus.OK)
                 .body(conversation);
     }
+
+    @GetMapping("/channels")
+    public List<BasicConversation> getAllChannels() {
+        return getBasicConversations(ConversationType.CHANNEL);
+    }
+
+    @GetMapping("/direct-messages")
+    public List<BasicConversation> getAllDirectMessages() {
+        return getBasicConversations(ConversationType.DIRECT_MESSAGE);
+    }
+
+    private List<BasicConversation> getBasicConversations(ConversationType conversationType) {
+        Iterable<Conversation> channelsIterable = conversationsRepository.findAll();
+        List<Conversation> channels = (List<Conversation>) channelsIterable;
+        return channels.stream()
```

Comment:
you can add method `findByConversationType(ConversationType type)` to `conversationRepository` and use it here :) you even don't need to implement it, Spring will generate sql itself basing on method name